sdk/core/azure-core/test/ut/curl_connection_pool_test.cpp

```diff
@@ -617,17 +636,38 @@ namespace Azure { namespace Core { namespace Test {
       auto connection
           = CurlConnectionPool::g_curlConnectionPool.ExtractOrCreateCurlConnection(req, options);
 
+      auto failedCounter = 0;
+      for (auto i = 0; i < _detail::AzureSdkHttpbinRetryCount; i++)
       {
-        // Check we can use the connection to retrieve headers in the response, but the connection
-        // is still flagged as shutdown.
-        auto session
-            = std::make_unique<Azure::Core::Http::CurlSession>(req, std::move(connection), options);
+        GTEST_LOG_(INFO) << "forceConnectionClosed test iteration " << i << ".";
```

Comment:
All of these appear to be essentially identical. Is it possible to centralize this logic in a single function that encapsulates the retry logic and includes the test-specific behavior in a lambda?

Reply:
I want to centralize the logic, but I prefer tests to be very simple/isolated and find the readability of lambdas a bit cognitively burdensome. If I can't find a simpler way to achieve that, will fall back to the lambda approach.